refactor: Sorting and remove code/__HELPERS/unsorted.dm

littleboobs opened the pull request:
## Что этот ПР делает
Отсортирован и удален `code/__HELPERS/unsorted.dm`
Общая ревизия всего `code/__HELPERS/`
ТГ база
- ДМДОК
- Кодстайл
- Некоторые небольшие оптимизации проков, не требующие рефакторов, например, замена Сишных лупов на обычные.
- Снейккейс некоторых проков.
- Некоторые улучшения манипуляций с JSON стрингами, дефайны джейсонов.
- Удалены неиспользуемые проки, какие-то остались от рефакторов, какие-то просто неиспользуемое легаси. Объединены проки с разным неймингом, копипастнутые кем-то с ТГ, в то время как был легаси аналог не со снейккейс названием.

Вроде бы основные моменты пометил. Оставил какое-то кол-во марков, где нужно портировать рефакторы, чтобы не пихать их в этот пр.

littleboobs committed:
del legacy AnimationLibrary from Goon and refactor/bugfix 'prayer bread'

Удаление неиспользуемой легаси либы с Гунов, давностью 10 лет. Была использована только в 3-х местах. Использованные проки, были перенесены локально туда, где они использовались. Рефактор использованных проков и сломанного 'prayer bread', пр указан ниже.

- https://github.com/ParadiseSS13/Paradise/pull/23056

littleboobs committed:
sorting, remove unused procs, dmdoc, naming vars, some optimization

linter fix

dageavtobusnick commented on code/__DEFINES/dmjit.dm:
```diff
@@ -45,9 +45,6 @@
 /proc/dmjit_hook_compile()
 	DMJIT_NATIVE
 
-/proc/auxtools_stack_trace(msg)
```
Верни на родину. Это скорее всего хук для dmjit и он нужен

littleboobs replied:
он в code/__HELPERS/_auxtools_api.dm
<img width="334" height="194" alt="image" src="https://github.com/user-attachments/assets/ffa45024-0e78-4dd4-acdc-fb64ae398b76" />

dageavtobusnick commented on code/__HELPERS/tool_helpers.dm:
```diff
@@ -1,57 +0,0 @@
-	istype(W, /obj/item/bonesetter)
-	)
-
-/proc/is_surgery_tool_by_behavior(obj/item/W)
```
Это бы оставить, может быть полезно, только заменить на is_surgery_tool, заместо кала чуть выше.

littleboobs replied:
Все ис хелперы из файла, что использовались, перенёс в макросы. Всё, кроме iscoil используется по 1 разу в каких-то легаси местах. Все эти легаси проки-хелперы с Парадиза. Конкретно is_surgery_tool_by_behavior там  используется в их атак чейне, его если кто-то и будет использовать, то вероятно при портировании атак чейна, в чего не будет. Могу оставить, но на деле нет смысла в ни разу не используемом макросе, проке, хелпере и т.д., как показывает практика.

dageavtobusnick replied:
Оставь. Только нейм смени, как я сказал. Это может быть полезно в будущем, в отличае от is_surgery_tool, который просто хуже

littleboobs replied:
Окей

dageavtobusnick commented on code/__HELPERS/AnimationLibrary.dm:
```diff
@@ -1,191 +0,0 @@
-	var/original_y = A.pixel_y
-	animate(A, transform = matrix(punchstr, MATRIX_ROTATE), pixel_y = 16, time = 2, color = "#eeeeee", easing = BOUNCE_EASING)
-	animate(transform = matrix(-punchstr, MATRIX_ROTATE), pixel_y = original_y, time = 2, color = "#ffffff", easing = BOUNCE_EASING)
-	animate(transform = null, time = 3, easing = BOUNCE_EASING)
```
Оставь. Тут вроде неплохие эффекты. Особенно для джунов. Либу нужно просто немного почистить и задокументировать.

littleboobs replied:
Мне не хочется, чтобы их использовали пельмени и т.д. Анимации базовые и не плохие, но скорее, как референс, чем реальные проки, что следует использовать?

dageavtobusnick replied:
Пусть будут. Это лучше, чем если пельмени будут делать свое и говнокодить.

dageavtobusnick commented on code/__HELPERS/_lists.dm:
```diff
@@ -322,44 +536,44 @@
-	return null
-
-//Pick a random element by weight from the list and remove it from the list.
-/proc/pick_weight_n_take(list/listfrom)
```
Потерял пик по весам

littleboobs replied:
Это какой-то непонятный неиспользуемый прок от Димача, видимо родившийся в копипасте. Там выше 2 используемых прока с пиком по весам.

dageavtobusnick replied:
Да, а этот нужен, чтобы удалять пикнутый по весам объект, как удаояется без весов выше. Он может быть полезен.

littleboobs replied:
Я умею читать, но за 3 года он так никому и не пригодился. Даже автору.

dageavtobusnick replied:
Но тем-неменее это хороший прок, который нет нужды удалять.

littleboobs replied:
Ладно

dageavtobusnick commented on code/__HELPERS/game.dm:
```diff
@@ -84,80 +98,39 @@
 		if(is_mob || isobj(thing))
 			. |= recursive_mob_check(thing, ., 3, include_clientless, include_radio, FALSE)
 
-/proc/get_cardinal_step_away(atom/start, atom/finish) //returns the position of a step from start away from finish, in one of the cardinal directions
```
как будто бы стоит оставить

littleboobs replied:
Это легаси прок, который когда-то использовался где-то, но был заменен на другой и код и остался здесь. Нет смысла хранить весь этот кал, который никто использовать, кроме пельменей, в реализации какого-то плохого кода, не будет. Так делал Фурукай, он копипастил с ТГ проки, использовал какое-то неиспользуемое легаси говно ещё с Бей времён и всё превращалось в кашу. Он, например, использовал какой-то палеозойский Гет_Ареа, вместо встроенной функции бульона.
Без референса использования любой неиспользуемый прок - щит.

dageavtobusnick replied:
Не все проки с бея и легаси проки плохие. Про гет_ареа я ничего не говорю. Это говно.

littleboobs replied:
Это то ладно, этот прок зачем?

dageavtobusnick replied:
Я проверил, в целом незачем. Он даже в старом говне, по типу vg не юзается. По идее у него в коде есть альтернатива, так что можешь удалять.

dageavtobusnick commented on code/__HELPERS/game.dm:
```diff
@@ -231,21 +204,6 @@
 /area/flick_overlay_view(mutable_appearance/display, duration)
 	return
 
-/proc/get_active_player_count()
```
И это, если ему нету альтернативы

littleboobs replied:
Он нигде не используется, даже на офах, какой-то легаси прок, вытесненный чем-то другим. Тебе хочется, чтобы это кто-то использовал? У нас есть места, где выводится подсчёт разных игроков, от гостов, до активных, нигде оно не используется.
<img width="701" height="285" alt="image" src="https://github.com/user-attachments/assets/bc2276ba-b05a-4324-9158-71e82c4318ea" />

dageavtobusnick replied:
Это можешь тогда удалять.

dageavtobusnick commented on code/__HELPERS/mobs.dm:
```diff
@@ -565,53 +546,11 @@
-/mob/verb/MouseSubstitute(params as command_text)
-	set hidden = 1
-	set name = ".mouse"
-	LogMouseMacro(".mouse", params)
```
Эти проки стоит оставить. Они нужны чтобы палить макросников.

littleboobs replied:
Не знал, верну

dageavtobusnick commented on code/__HELPERS/mobs.dm:
```diff
@@ -565,53 +546,11 @@
-	var/list/view_range = getviewsize(user.client.view)
-	var/turf/source = get_turf(user)
-	var/turf/target = get_turf(A)
-	return ISINRANGE(target.x, source.x - view_range[1], source.x + view_range[1]) && ISINRANGE(target.y, source.y - view_range[1], source.y + view_range[1])
```
Ты переместил эти проки или удалил, если второе - верни

littleboobs replied:
Какие использовались или используются на ТГ - переместил. То что было в тупую копипастнуто с офов или неиспользуемое легаси - удалил. Конкретно in_view_range используется в компоненте автоогня, его копипастнул Звай и откзался от него в своей реализации. На ТГ используется макрос CAN_THEY_SEE. У нас не используется ничего.

dageavtobusnick replied:
Хорошо, тогда можешь удалять.

dageavtobusnick commented on code/__HELPERS/text.dm:
```diff
@@ -192,61 +168,21 @@
-/proc/dd_hassuffix_case(text, suffix)
-	var/start = length(text) - length(suffix)
-	if(start)
-		return findtextEx_char(text, suffix, start, null)
```
все детекторы префиксов и суффиксов нужно оставить.

littleboobs replied:
Не нужно, это легаси кал, который с каждым рефактором вычищается, я пометил это и вычищу потом, то что используется.

dageavtobusnick replied:
Ладно, похуй. В каком-то смысле тут ты прав, но наверное даже сам не понимаешь - в чем.

dageavtobusnick commented on code/__HELPERS/text.dm:
```diff
@@ -255,40 +191,20 @@
-	return list(copytext(t, 1, 2), copytext(t, 2, 0))
-
-//Strips the last char and returns it and the new string as a list
-/proc/strip_last(t)
```
и это тоже, но в виде макросов

littleboobs replied:
Зачем?

dageavtobusnick replied:
Это полезно, чтобы отрезать первый или последний символ от строки. Но проки тут - лишнее, хватит и макроса.

littleboobs replied:
Везде, где это нужно используются реплейсы и копитексты напрямую по нужде. Кому и зачем нужен макрос/прок, который нигде не используется и никому не нужен, особенно учитывая то, как редко у нас кто-то самостоятельно работает со стрингами, а не копипастит с другого билда.
<img width="772" height="120" alt="image" src="https://github.com/user-attachments/assets/9eeb884b-e04a-4127-8b1f-9579a9b6ee38" />
<img width="827" height="107" alt="image" src="https://github.com/user-attachments/assets/f9ffeb7d-1118-4917-a09a-d3d95c8d6d6d" />

dageavtobusnick replied:
Но тем-неменее они имеют право на существования в виде макросов.

dageavtobusnick commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1,1919 +0,0 @@
-	tX = tX[1]
-	tX = max(1, min(world.maxx, origin.x + (text2num(tX) - (world.view + 1))))
-	tY = max(1, min(world.maxy, origin.y + (text2num(tY) - (world.view + 1))))
-	return locate(tX, tY, tZ)
```
Это бы вернуть, если удалил

littleboobs replied:
Зачем он тебе? Он не используется ни у нас, ни на ТГ. На ТГ такой комментарий.
<img width="625" height="276" alt="image" src="https://github.com/user-attachments/assets/d190d43a-c93e-4fa0-91b0-bb6c765e61c1" />

dageavtobusnick replied:
тогда можешь удалять

dageavtobusnick commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1,1919 +0,0 @@
-/proc/topic_link(datum/D, arglist, content)
-	if(istype(arglist,/list))
-		arglist = list2params(arglist)
-	return "<a href='byond://?src=[D.UID()];[arglist]'>[content]</a>"
```
А это выглядит как нихуевая идея для рефактора, но только макросом. Но а пока да, не нужно.

littleboobs replied:
Ну да, ТГ
<img width="1388" height="61" alt="image" src="https://github.com/user-attachments/assets/d6f67c6a-1b29-4bda-b32a-d19d19a7bff0" />

dageavtobusnick commented on code/__HELPERS/atoms.dm:
```diff
@@ -155,3 +156,498 @@
+		return FALSE
+	if(item.sharp)
+		return TRUE
+	return FALSE
```
можно переделать в макрос

littleboobs replied:
там не макрос, а переменная шарпа у итема есть, не стал в этом пр-е делать

dageavtobusnick commented on code/__HELPERS/atoms.dm:
```diff
@@ -155,3 +156,498 @@
+		return TRUE
+
+	if(is_type_in_typecache(item, GLOB.can_embed_types))
+		return TRUE
```
и тут

dageavtobusnick commented on code/__HELPERS/view.dm:
```diff
@@ -1,6 +1,52 @@
+			. = view(distance,center)
+		if("range")
+			. = range(distance,center)
+	return
```
Этот кал можно и нужно удалить

littleboobs replied:
этот кал используется))

dageavtobusnick replied:
Ну вот его надо убрать. Это в худшем случае не дает ничего, а в лудшем убивает внутрений оптимизон в for

littleboobs replied:
там какой-то кал со спелл системой офов, я не хочу трогать
<img width="338" height="71" alt="image" src="https://github.com/user-attachments/assets/791fcf7c-aec6-4b84-9326-3039b19e078b" />
<img width="530" height="46" alt="image" src="https://github.com/user-attachments/assets/92ad3d0c-4b9b-4d05-bb51-de50356044e4" />

dageavtobusnick reviewed (changes requested):
Может ты и не удалил указанные проки, но из-за того, что в один пр смешано 4-5 я просто не смог их найти

littleboobs committed:
dage review

- can_embed переделан макрос.
- Избыточный is_sharp заменён на прямое использование переменной sharp у итема.
- Возвращена аним либа. Библа задокументирована и отрефакторена.
- hasorgans заменён на iscarbon.
- Макрос is_surgery_tool, который раньше был проком is_surgery_tool_by_behavior. Нигде не используется.

littleboobs commented:
Поправил всё, кроме открытых комментариев.

dageavtobusnick commented on code/modules/events/spacevine.dm:
```diff
@@ -349,7 +349,7 @@
 
 /datum/spacevine_mutation/woodening/on_hit(obj/structure/spacevine/holder, mob/living/hitter, obj/item/item, expected_damage)
-	if(!is_sharp(item))
+	if(!item.sharp)
```
Смотри. is_sharp еще проверял на соответствик типу, что отсекало null и объекты неправильного типа. Сейчас же у тебя может отвалиться многое.

littleboobs replied:
Знаю, там везде явно тип указан obj/item/I но сейчас глазами пробегусь ещё раз

dageavtobusnick replied:
И что, что указан? Если нет проверки istype, это никак не защищает

dageavtobusnick replied:
Там можно хоть число передать

dageavtobusnick replied:
Это не шарпы, джава или раст, где компилятор такое не позволит

littleboobs replied:
В смысле? По такой логике каждый прок с аргументами должен начинаться с тонны истайп проверок. Мб я че то не понял или не понимаю

littleboobs replied:
Если такой прок с явным указанием типа аргумента, то зачем нужен истайп?
```
/obj/item/fish/salmon/attackby(obj/item/I, mob/user, params)
	if(I.sharp)
```

dageavtobusnick replied:
Бля. Я хз как это текстом грамотно объяснить.

littleboobs committed:
dage review 2
